pkg/router/router.go

```diff
@@ -26,50 +27,150 @@ import (
 	restful "github.com/emicklei/go-restful"
 	"github.com/tektoncd/dashboard/pkg/endpoints"
 	logging "github.com/tektoncd/dashboard/pkg/logging"
-	"github.com/tektoncd/dashboard/pkg/utils"
 	corev1 "k8s.io/api/core/v1"
 )
 
-// Label key required by services to be registered as a dashboard extension
+// ExtensionLabelKey is the label key required by services to be registered as a
+// dashboard extension
 const ExtensionLabelKey = "tekton-dashboard-extension"
 
-// Label value required by services to be registered as a dashboard extension
+// ExtensionLabelValue is the label value required by services to be registered
+// as a dashboard extension
 const ExtensionLabelValue = "true"
 
-// Full label required by services to be registered as a dashboard extension
+// ExtensionLabel is the full label required by services to be registered as a
+// dashboard extension
 const ExtensionLabel = ExtensionLabelKey + "=" + ExtensionLabelValue
 
-// Annotation that specifies the valid extension path, defaults to "/"
-const ExtensionUrlKey = "tekton-dashboard-endpoints"
+// ExtensionURLKey specifies the valid extension paths, defaults to "/"
+const ExtensionURLKey = "tekton-dashboard-endpoints"
 
-// Delimiter to be used between the extension endpoints annotation value
+// ExtensionEndpointDelimiter is the Delimiter to be used between the extension
+// endpoints annotation value
 const ExtensionEndpointDelimiter = "."
 
-// Extension UI bundle location annotation
+// ExtensionBundleLocationKey IS the UI bundle location annotation key
 const ExtensionBundleLocationKey = "tekton-dashboard-bundle-location"
 
-// Extension display name annotation
+// ExtensionDisplayNameKey is the display name annotation key
 const ExtensionDisplayNameKey = "tekton-dashboard-display-name"
 
-// extensionRoot
+// ExtensionRoot is the URL root when accessing extensions
 const ExtensionRoot = "/v1/extensions"
 
 var webResourcesDir = os.Getenv("WEB_RESOURCES_DIR")
 
-func Register(resource endpoints.Resource) *restful.Container {
-	wsContainer := restful.NewContainer()
-	wsContainer.Router(restful.CurlyRouter{})
-
+// Register returns an HTTP handler that has the Dashboard REST API registered
+func Register(resource endpoints.Resource) *Handler {
 	logging.Log.Info("Registering all endpoints")
-	registerWeb(wsContainer)
-	registerEndpoints(resource, wsContainer)
-	registerPropertiesEndpoint(resource, wsContainer)
-	registerWebsocket(resource, wsContainer)
-	registerHealthProbe(resource, wsContainer)
-	registerReadinessProbe(resource, wsContainer)
-	registerExtensions(wsContainer)
-	registerKubeAPIProxy(resource, wsContainer)
-	return wsContainer
+	h := &Handler{
+		Container:       restful.NewContainer(),
+		uidExtensionMap: make(map[string]*Extension),
```

Comment:
We definitely need the *?

Reply:
The old function definition was
```go
func Register(resource endpoints.Resource) *restful.Container
```
so I continued to use that pattern.

Reply:
👍